src/metal.cs

```diff
@@ -2845,8 +3095,39 @@ interface MTLArgumentEncoder
 		[Export ("constantDataAtIndex:")]
 		IntPtr GetConstantData (nuint index);
 
-		[NoTV][NoiOS]
+		[NoiOS, NoTV]
+		[Mac (10,14, onlyOn64: true)]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("setRenderPipelineState:atIndex:")]
+		void SetRenderPipelineState ([NullAllowed] IMTLRenderPipelineState pipeline, nuint index);
+
+		[NoiOS, NoTV]
+		[Mac (10,14, onlyOn64: true)]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("setRenderPipelineStates:withRange:")]
+		void SetRenderPipelineStates (IMTLRenderPipelineState[] pipelines, NSRange range);
+
+		[Mac (10,14, onlyOn64: true)]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("setIndirectCommandBuffer:atIndex:")]
+		void SetIndirectCommandBuffer ([NullAllowed] IMTLIndirectCommandBuffer indirectCommandBuffer, nuint index);
```

Comment:
Same as the above set methods, IndirectCommandBuffer can be deduced from the first arg.